app/questionnaire/questionnaire_schema.py

```diff
@@ -839,17 +882,70 @@ def _populate_when_rules_section_dependencies(self) -> None:
             when_rules = self.get_values_for_key(section, "when")
             rules: list = list(when_rules)
 
-            if rules_section_dependencies := self._get_rules_section_dependencies(
-                section["id"], rules
-            ):
+            (
+                rules_section_dependencies,
+                rule_section_dependencies_for_progress_value_source,
+                rule_block_dependencies_for_progress_value_source,
+            ) = self._get_rules_section_dependencies(section["id"], rules)
+
+            if rules_section_dependencies:
                 self._when_rules_section_dependencies_by_section[
                     section["id"]
                 ] = rules_section_dependencies
 
+            if rule_section_dependencies_for_progress_value_source:
+                self._when_rules_section_dependencies_by_section_for_progress_value_source[
+                    section["id"]
+                ] = rule_section_dependencies_for_progress_value_source
+
+            if rule_block_dependencies_for_progress_value_source:
+                self._when_rules_block_dependencies_by_section_for_progress_value_source[
+                    section["id"]
+                ] = rule_block_dependencies_for_progress_value_source
+
+    def _get_section_and_block_ids_dependencies_for_progress_source_and_answer_ids_from_rule(
+        self, current_section_id: str, rule: Mapping
+    ) -> tuple[set[str], dict[str, set[str]]]:
+        answer_id_list: set[str] = set()
+        dependencies_ids_for_progress_value_source: dict[str, set[str]] = {
+            "sections": set(),
+            "blocks": set(),
+        }
+        identifier: Optional[str] = rule.get("identifier")
+        source: Optional[str] = rule.get("source")
+        selector: Optional[str] = rule.get("selector")
+
+        if source == "answers" and identifier:
+            answer_id_list.add(identifier)
+        elif source == "calculated_summary" and identifier:
```

Comment:
The `and` isn't needed as `identifier` is mandatory.
```suggestion
        if source == "answers":
            answer_id_list.add(identifier)
        elif source == "calculated_summary":
```

Reply:
As above